### PR TITLE
Add gpt-4o model

### PR DESCRIPTION
**Pull request by err09r**
Source: https://platform.openai.com/docs/models/gpt-4o

### CODE REVIEW

File: bot/openai_helper.py [hunk 1]
```diff
@@ -28,7 +28,8 @@
 GPT_4_32K_MODELS = ("gpt-4-32k", "gpt-4-32k-0314", "gpt-4-32k-0613")
 GPT_4_VISION_MODELS = ("gpt-4-vision-preview",)
 GPT_4_128K_MODELS = ("gpt-4-1106-preview","gpt-4-0125-preview","gpt-4-turbo-preview")
-GPT_ALL_MODELS = GPT_3_MODELS + GPT_3_16K_MODELS + GPT_4_MODELS + GPT_4_32K_MODELS + GPT_4_VISION_MODELS + GPT_4_128K_MODELS
+GPT_4O_MODELS = ("gpt-4o")
```
Comment (lucasaxm) [inline bot/openai_helper.py:31]: You missed the comma to create a tuple with one item
```
TypeError: can only concatenate tuple (not "str") to tuple
Traceback (most recent call last):
  File "/app/bot/main.py", line 7, in <module>
    from openai_helper import OpenAIHelper, default_max_tokens, are_functions_available
  File "/app/bot/openai_helper.py", line 32, in <module>
    GPT_ALL_MODELS = GPT_3_MODELS + GPT_3_16K_MODELS + GPT_4_MODELS + GPT_4_32K_MODELS + GPT_4_VISION_MODELS + GPT_4_128K_MODELS + GPT_4O_MODELS
```

```suggestion
GPT_4O_MODELS = ("gpt-4o",)
```

**Review by lucasaxm — CHANGES_REQUESTED** [review]
Bump up version of `tiktoken` from `0.5.1` to `0.7.0` which supports `gpt-4o`

https://github.com/openai/tiktoken/blob/main/CHANGELOG.md#v070

**Comment by LesnoyChelovek** [comment]
Not workL

`2024-05-14 06:04:03,896 - root - ERROR - ⚠ _An error has occurred._ ⚠
Unknown encoding gpt-3.5-turbo. Plugins found: ['tiktoken_ext.openai_public']
Traceback (most recent call last):
  File "/home/ubuntu/chatgpt-telegram-bot/bot/openai_helper.py", line 650, in __count_tokens
    encoding = tiktoken.encoding_for_model(model)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/tiktoken/model.py", line 97, in encoding_for_model
    return get_encoding(encoding_name_for_model(model_name))
  File "/home/ubuntu/.local/lib/python3.10/site-packages/tiktoken/model.py", line 84, in encoding_name_for_model
    raise KeyError(
KeyError: 'Could not automatically map gpt-4o to a tokeniser. Please use `tiktoken.get_encoding` to explicitly get the token
iser you expect.'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/chatgpt-telegram-bot/bot/openai_helper.py", line 231, in __common_get_chat_response
    token_count = self.__count_tokens(self.conversations[chat_id])
  File "/home/ubuntu/chatgpt-telegram-bot/bot/openai_helper.py", line 652, in __count_tokens
    encoding = tiktoken.get_encoding("gpt-3.5-turbo")
  File "/home/ubuntu/.local/lib/python3.10/site-packages/tiktoken/registry.py", line 68, in get_encoding
    raise ValueError(
ValueError: Unknown encoding gpt-3.5-turbo. Plugins found: ['tiktoken_ext.openai_public']

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ubuntu/chatgpt-telegram-bot/bot/telegram_bot.py", line 697, in prompt
    async for content, tokens in stream_response:
  File "/home/ubuntu/chatgpt-telegram-bot/bot/openai_helper.py", line 178, in get_chat_response_stream
    response = await self.__common_get_chat_response(chat_id, query, stream=True)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/tenacity/_asyncio.py", line 88, in async_wrapped
    return await fn(*args, **kwargs)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/tenacity/_asyncio.py", line 47, in __call__
    do = self.iter(retry_state=retry_state)
  File "/home/ubuntu/.local/lib/python3.10/site-packages/tenacity/__init__.py", line 314, in iter
    return fut.result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/home/ubuntu/.local/lib/python3.10/site-packages/tenacity/_asyncio.py", line 50, in __call__
    result = await fn(*args, **kwargs)
  File "/home/ubuntu/chatgpt-telegram-bot/bot/openai_helper.py", line 272, in __common_get_chat_response
    raise Exception(f"⚠ _{localized_text('error', bot_language)}._ ⚠\n{str(e)}") from e
Exception: ⚠ _An error has occurred._ ⚠
Unknown encoding gpt-3.5-turbo. Plugins found: ['tiktoken_ext.openai_public']`

Снимок экрана 2024-05-14 в 09 05 26

**Review by muhammed540 — APPROVED** [review]

**Comment by bulat15g** [comment]
worked for me

**Review by bulat15g — APPROVED** [review]

**Comment by upstream-team** [comment]
perfect

**Comment by n3d1117** [comment]
Thanks everyone!

**Comment by gianlucaalfa** [comment]
I also update openai==1.3.3 to openai==1.29.0

**Comment by n3d1117** [comment]
> I also update openai==1.3.3 to openai==1.29.0

Done! Thanks

**Pull request merged**